.github/workflows/build.sh

```diff
@@ -20,7 +24,37 @@ case "$1" in
             export UBSAN_OPTIONS=print_stacktrace=1:print_summary=1:halt_on_error=1
         fi
 
-        ./bootstrap.sh --enable-tests --prefix=/usr
+        if [[ "$WERROR" == true ]]; then
+            # acx_pthread isn't compatible with CFLAGS=-Werror at all so it's just replaced
+            # with ax_pthread from autoconf-archive.
+            rm -rf common/acx_pthread.m4
+
+            export CFLAGS="-Werror"
```

Comment:
I would use CFLAGS+= here too and below all changes just export CFLAGS. It would make possible configuring CFLAGS from yaml file.

Reply:
I'll replace `CFLAGS=` with `CFLAGS+=` but I would keep passing `WERROR` and `UBSAN_UBSAN` instead of `CFLAGS`: https://github.com/lathiat/avahi/pull/425#discussion_r1044925323 should be applicable here too.

Reply:
I replaced `CFLAGS=` with `CFLAGS+=` and started using it to pass various optimization levels to catch even more warnings.